### PR TITLE
847 connector plugin docs

### PR DESCRIPTION
**Pull request by izgeri**
#### What does this PR do (include background context, if relevant)?
This PR updates the in-repo docs to include up-to-date information about the new plugin interface based on the [plugin README](https://gist.github.com/jonahx/6892ddeaca03f6a5c281eb38218687eb#connector-resources---argument-passed-to-your-constructor) written during the design phase of #693. Including in this work are:
- Updated CONTRIBUTING file
- Minor updates to old, now internal plugin README
- New godoc overviews for the new plugin, connector plugin, and log packages
- New README files in the plugin / connector plugin folders
- Minor adjustments to comments in some of the public packages to better capture content originally available in the plugin README

#### What ticket does this PR close?
Connected to #847 

#### Where should the reviewer start?
- You can view the godocs locally by running `./bin/view_godocs` and navigating to http://localhost:6060/pkg/secretless-broker/ in your browser
- PDFs of the relevant godocs are also available here: [Secretless Plugin Godocs.zip](https://github.com/cyberark/secretless-broker/files/3745084/Secretless.Plugin.Godocs.zip)
  These PDFs are primarily capturing godoc content available in the `log` and `plugin` parts of the doc tree:
  Screen Shot 2019-10-18 at 1 40 53 PM
- You can view the files in the "files changed" section of the PR to preview how the markdown / README changes will look in github

### CODE REVIEW

File: pkg/secretless/plugin/connector/README.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+  Error(msg string)
+  Errorln(msg string)
+
+  Fatalf(format string, v ...interface{})
+  Fatal(msg string)
+  Fatalln(msg string)
+}
+```
+
+The 3 `Debug` methods and the 3 `Info` methods do nothing unless you started
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: The three 'Debug' methods and the three 'Info' methods....

File: pkg/secretless/plugin/connector/README.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+Secretless with the `Debug` command line flag set to true.  If you did start
+Secretless in debug mode, they write to stdout.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: If you did start Secretless in debug mode, the methods write to stdout.

File: pkg/secretless/plugin/connector/README.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+The `Warn`, `Error`, and `Fatal` methods all write to stderr, regardless of
+the current `Debug` mode.  
+
+All the `Fatal` methods call `os.Exit(1)` after printing their message.
+
+The `Logger` automatically prepends the name of the currently running service
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: currently-running service....

File: pkg/secretless/plugin/connector/README.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+to all messages.  That is, the service name specified in your `secretless.yml`.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: ....to all messages, that is the service name.......

File: pkg/secretless/plugin/connector/README.md [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+```
+version: "v2"
+services:
+  sample-service:
+    protocol: pg
+    listenOn: unix:///sock/.s.PGSQL.5432
+    ...
+```
+
+then the `Logger` will prepend `sample-service` to all messages.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: then the 'Logger' prepends 'sample-service' to all messages.

File: pkg/secretless/plugin/connector/README.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+
+## Examples
+
+To see an example external connector, please take a look at our [test plugin](../../../../test/plugin).
+You can also look at our internal plugins (which also implement this interface):
```
Comment (suefran) [inline pkg/secretless/plugin/connector/README.md]: You can also look at our internal plugins that also implement this interface. 

Not sure why some of the sentence was in parens?

File: pkg/secretless/plugin/connector/doc.go [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/*
+
+Package connector enables you to extend the Secretless functionality beyond
```
Comment (suefran) [inline pkg/secretless/plugin/connector/doc.go]: I'd rephrase to:

Package connector allows you to extend the functionality of Secretless beyond currently-supported internal plugins."

File: pkg/secretless/plugin/connector/doc.go [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+the natively supported internal connector plugins.
+
+For more detailed information on connector plugin architecture and how to get started
```
Comment (suefran) [inline pkg/secretless/plugin/connector/doc.go]: For more information on the connector.....

File: pkg/secretless/plugin/connector/doc.go [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+building connector plugins, please refer to the README.md file in this directory in GitHub, eg
```
Comment (suefran) [inline pkg/secretless/plugin/connector/doc.go]: don't think you need the "eg." I'd just include the url.

File: pkg/secretless/plugin/connector/http/http.go [hunk 10]
```diff
@@ -15,18 +15,28 @@ type Plugin interface {
...
-// function that knows how to authenticate an http request. It uses
-// its credential values argument passed in to modify the http.Request
-// request object and the credentials map to authenticate the client.
+/*
+Connector is an interface with a single method, Connect. Connect is a
+function that knows how to authenticate an http request. It uses
+its credential values argument to modify the http.Request and authenticate the
+client.
+
+Typically altering the HTTP request so that it contains the necessary
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: Typically, altering....

File: pkg/secretless/plugin/connector/http/http.go [hunk 11]
```diff
@@ -15,18 +15,28 @@ type Plugin interface {
 	NewConnector(connector.Resources) Connector
 }
 
-// Connector is an interface with a single method, "Connect".  "Connect" is a
-// function that knows how to authenticate an http request. It uses
-// its credential values argument passed in to modify the http.Request
-// request object and the credentials map to authenticate the client.
+/*
+Connector is an interface with a single method, Connect. Connect is a
+function that knows how to authenticate an http request. It uses
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: You've used a mix of 'http' and 'HTTP." I'd pick one and be consistent.

File: pkg/secretless/plugin/connector/http/http.go [hunk 12]
```diff
@@ -15,18 +15,28 @@ type Plugin interface {
...
+its credential values argument to modify the http.Request and authenticate the
+client.
+
+Typically altering the HTTP request so that it contains the necessary
+authentication information means adding the appropriate headers to the request
+-- for example, an `Authorization` header containing a Token, or a header
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: I don't think "Token" should be uppercase unless you are referring to a specific token by name. If a more generic "token" lowercase is fine.

File: pkg/secretless/plugin/connector/http/http.go [hunk 13]
```diff
@@ -15,18 +15,28 @@ type Plugin interface {
...
+containing an API key.
+
+Since HTTP is a stateless protocol, Secretless will call the Connect function every
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: .....Secretless calls the...

File: pkg/secretless/plugin/connector/http/http.go [hunk 14]
```diff
@@ -15,18 +15,28 @@ type Plugin interface {
...
+time a client sends an HTTP request to the target server, so that every request
+will be authenticated.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: ......so that every request is authenticated.

File: pkg/secretless/plugin/connector/http/http.go [hunk 15]
```diff
@@ -36,10 +46,16 @@ type Connector interface {
 // and you now have a Plugin. It does this by giving any function of the type
 // ConnectorConstructor a constructor method called NewConnector that simply
 // calls the function itself
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: add a period.

File: pkg/secretless/plugin/connector/http/http.go [hunk 16]
```diff
@@ -36,10 +46,16 @@ type Connector interface {
...
-type ConnectorConstructor func (connector.Resources) Connector
+type ConnectorConstructor func(connector.Resources) Connector
+
+/*
+NewConnector returns a Connector, a one method interface that performs the actual
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: NewConnector returns a Connector, which is a one method interface.....

File: pkg/secretless/plugin/connector/http/http.go [hunk 17]
```diff
@@ -36,10 +46,16 @@ type Connector interface {
...
+authentication.
 
-// NewConnector is a ConnectorConstructor's implementation of the Plugin
-// interface. It simply delegates to the underlying function.
+When Secretless runs, it will call NewConnector once and then hold onto the
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: When Secretless runs, it calls...........once and then holds onto the.................

File: pkg/secretless/plugin/connector/http/http.go [hunk 18]
```diff
@@ -36,10 +46,16 @@ type Connector interface {
...
+returned Connector.  That Connector (remember: it's just a a single method)
+will then be called each time a new client connection requires authentication.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/http/http.go]: is then called each time a new........

File: pkg/secretless/plugin/connector/resources.go [hunk 19]
```diff
@@ -4,15 +4,34 @@ import (
 	"github.com/cyberark/secretless-broker/pkg/secretless/log"
 )
 
-// Resources is an interface that defines what things will
-// be needed by a plugin at runtime.
+/*
+Resources is an interface that exposes everything that your Connector needs from
```
Comment (suefran) [inline pkg/secretless/plugin/connector/resources.go]: Resources is an interface that exposes everything your Connector needs.......

File: pkg/secretless/plugin/connector/resources.go [hunk 20]
```diff
@@ -4,15 +4,34 @@ import (
...
+the Secretless framework and is passed to your plugin's
+constructor.  You will need to retain a reference to connector.Resources, via
```
Comment (suefran) [inline pkg/secretless/plugin/connector/resources.go]: You need to retain a........

File: pkg/secretless/plugin/connector/resources.go [hunk 21]
```diff
@@ -4,15 +4,34 @@ import (
...
+*/
 type Resources interface {
-	// Config returns the content of addition configuration
-	// parameters passed to the connector.
+	/*
+		Config() provides your connector with resources specified in your secretless.yml
+		file.
+
+		Anything specified in your connector's `config` section of the secretless.yml
+		will be passed back via this method as a raw []byte. Not all connectors require
```
Comment (suefran) [inline pkg/secretless/plugin/connector/resources.go]: is passed back via......

File: pkg/secretless/plugin/connector/resources.go [hunk 22]
```diff
@@ -4,15 +4,34 @@ import (
...
-	// Logger returns an instance of a Logger that can be used
-	// to record logging messages from the plugin.
+	/*
+		Logger() provides a basic logger you can use for debugging and informational
+		logging.
+
+		This method returns an object similar to the standard library's
+		`log.Logger` to let you log events to stdout and stderr.  It respects
+		Secretless's command line `debug` flag, so that calling Debugf or Infof
+		will do nothing unless you started Secretless in debug mode.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/resources.go]: .....does nothing unless you started......

File: pkg/secretless/plugin/connector/tcp/tcp.go [hunk 23]
```diff
@@ -15,9 +15,23 @@ type Plugin interface {
...
-// Connector is an interface with a single method, "Connect".  "Connect" is a
-// function that takes credentials and an unauthenticated connection and returns
-// an authenticated connection to the target service.
+/*
+Connector is an interface with a single method, Connect. Connect is a
+function that takes credentials and an unauthenticated connection and returns
+an authenticated connection to the target service.
+
+The authentication stage is complete after Connect is called. At that point,
+Secretless will have both the client connection and an authenticated
```
Comment (suefran) [inline pkg/secretless/plugin/connector/tcp/tcp.go]: At that point, Secretless has both the client connection........

File: pkg/secretless/plugin/connector/tcp/tcp.go [hunk 24]
```diff
@@ -49,10 +63,16 @@ func (cf ConnectorFunc) Connect(
...
-type ConnectorConstructor func (connector.Resources) Connector
+type ConnectorConstructor func(connector.Resources) Connector
+
+/*
+NewConnector returns a Connector, a one method interface that performs the actual
+authentication.
 
-// NewConnector is a ConnectorConstructor's implementation of the Plugin
-// interface. It simply delegates to the underlying function.
+When Secretless runs, it will call NewConnector once and then hold onto the
```
Comment (suefran) [inline pkg/secretless/plugin/connector/tcp/tcp.go]: When Secretless runs, it calls NewConnector once and then holds onto the.......

File: pkg/secretless/plugin/connector/tcp/tcp.go [hunk 25]
```diff
@@ -49,10 +63,16 @@ func (cf ConnectorFunc) Connect(
...
+returned Connector.  That Connector (remember: it's just a a single method)
+will then be called each time a new client connection requires authentication.
```
Comment (suefran) [inline pkg/secretless/plugin/connector/tcp/tcp.go]: is then called each time......

File: pkg/secretless/plugin/doc.go [hunk 26]
```diff
@@ -0,0 +1,15 @@
+/*
+
+Package plugin enables you to extend the Secretless functionality beyond
```
Comment (suefran) [inline pkg/secretless/plugin/doc.go]: I'd rephrase to:

Package plugin allows you to extend the functionality of Secretless beyond currently-supported internal plugins."

File: pkg/secretless/plugin/doc.go [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+the natively supported internal plugins.
+
+At current, the plugin package supports the following plugin types:
```
Comment (suefran) [inline pkg/secretless/plugin/doc.go]: Currently, the plugin package supports the following types of plugins:

File: pkg/secretless/plugin/doc.go [hunk 28]
```diff
@@ -0,0 +1,15 @@
...
+
+  - Service connector plugins
+
+For more detailed information on the plugin architecture and how to get started
+building plugins, please refer to the README.md file in this directory in GitHub, eg
```
Comment (suefran) [inline pkg/secretless/plugin/doc.go]: remove eg

File: pkg/secretless/log/doc.go [hunk 29]
```diff
@@ -0,0 +1,35 @@
+/*
+Package log provides a type similar to the standard library's "log.Logger".  It allows you to
+log events to stdout and stderr.  It respects Secretless's command line `debug` flag,
+so that calling Debugf or Infof will do nothing unless you started Secretless in
```
Comment (suefran) [inline pkg/secretless/log/doc.go]: so that calling Debugf or Infof does nothing unless you......

File: pkg/secretless/log/doc.go [hunk 30]
```diff
@@ -0,0 +1,35 @@
...
+
+  version: "v2"
+  services:
+    sample-service:
+      protocol: pg
+      listenOn: unix:///sock/.s.PGSQL.5432
+      ...
+
+
+then the Logger will prepend "sample-service" to all messages.
```
Comment (suefran) [inline pkg/secretless/log/doc.go]: then the Logger prepends.....

File: pkg/secretless/plugin/README.md [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+  - [Service Connectors](connector/README.md)
+
+## External Plugin Basics
+
+Secretless uses [Go plugins](https://golang.org/pkg/plugin/).
+
+If you've never used Go plugins before, a good introduction to them is
+[here](https://medium.com/learning-the-go-programming-language/writing-modular-go-programs-with-plugins-ec46381ee1a9).
+The short version is that you'll write normal Go functions, but compile them
+with different flags.  This will produce a shared object library file (with a
```
Comment (suefran) [inline pkg/secretless/plugin/README.md]: This produces a shared......

File: pkg/secretless/plugin/README.md [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
+`.so` extension) instead of a normal executable.
+
+Technically, a Secretless plugin is just a Go shared library
+file that implements some predefined functions. For more information on what exactly
```
Comment (suefran) [inline pkg/secretless/plugin/README.md]: For more information on what you need to implement.......

File: pkg/secretless/plugin/README.md [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+- `id`: A short, clear, unique name, for use in logs and the `secretless.yml`
+  config file.  Allowed characters are: lowercase letters, `_`, `:`, `-`, and
+  `~`.
+- `description`: A short summary of the plugin, not to exceed 100 characters.
+  This may be used in the future by the Secretless command line tool to list
+  available plugins.
+
+## External Plugin Basics
+
+When running Secretless with external plugins, you may leverage some special
```
Comment (suefran) [inline pkg/secretless/plugin/README.md]: you can leverage....

File: CONTRIBUTING.md [hunk 34]
```diff
@@ -314,17 +314,9 @@ replace `<GOOS>/<GOARCH>` with your particular operating system and compilation
...
-- Handler plugins
-- Connection management plugins
-
-Currently, these API definitions reside [here](internal/plugin/v1) and an example plugin can be found in the [`test/plugin`](test/plugin) directory.
-
-You can read more about how to make plugins and the underlying architecture in the [API directory](internal/plugin).
-
-_Please note: Plugin API interface signatures and supported plugin API version(s) are currently under heavy development so they will be likely to change in the near future._
+Secretless supports using [Go plugins](https://golang.org/pkg/plugin/) to extend
+its functionality. To learn more about writing Secretless plugins and which plugins
```
[the remark this answers is not in the document]
Comment (izgeri) [reply CONTRIBUTING.md]: Editing to
> To learn about writing new Secretless plugins and for more information on the types of plugins we currently support, visit.....

Does that make sense? Also, lol @ "no please" :)

[the remark this answers is not in the document]
Comment (suefran) [reply CONTRIBUTING.md]: yes, excellent!

File: internal/plugin/README.md [hunk 35]
```diff
@@ -155,7 +155,7 @@ change in the near future_
 ## EventNotifier
 
 `EventNotifier` is used as a target object of events for handlers and listeners that notifies the plugin manager
-in an abstract way without needing to pass down the full connection manager as a parameter. 
+in an abstract way without needing to pass down the full connection manager as a parameter.
```
[the remark this answers is not in the document]
Comment (izgeri) [reply internal/plugin/README.md]: In general I think we agreed as a group that the EventNotifier construct was also confusing. It may be that our lack of a clear group understanding of it led to an unclear description as well. This is not part of the new interface, though an updated iteration of it may reappear when/if we add a plugin to support managing connections again.

File: pkg/secretless/plugin/README.md [hunk 36]
```diff
@@ -0,0 +1,94 @@
+# Secretless Plugins
+
+Secretless plugins provide a mechanism for extending the functionality of
+Secretless beyond the natively supported built-in plugins.
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/README.md]: How about
> Secretless plugins allow you to extend the functionality of Secretless beyond the currently supported built-in plugins.

There were a lot of conversations about internal vs built-in on slack, and the consensus seemed to be that built-in was preferred. I am trying to use that everywhere.

File: pkg/secretless/plugin/README.md [hunk 37]
```diff
@@ -0,0 +1,94 @@
...
+### Building the Shared Library File
+To build your plugin's shared library (`.so`) file, you follow the [instructions](https://golang.org/pkg/plugin/) for building Go plugins. That is, you run a command like:
+```
+go build -buildmode=plugin -o=/path/to/my-plugin.so my_plugin.go
+```
+
+This compiles your plugin code into a `.so` file. Once you've done this,
+you'll place the `.so` file in the `/usr/local/lib/secretless` directory in the
+container where Secretless will run (or in another directory you specify using the `-p` flag).
+That's all you have to do. The `.so` files are self-contained, with any dependencies
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/README.md]: I'm fine with making this change here, but I agree it would be good to discuss where it's appropriate to use a conversational tone - maybe it's appropriate in tutorials and blog posts, but not in reference material or guides?

File: pkg/secretless/plugin/connector/README.md [hunk 38]
```diff
@@ -0,0 +1,267 @@
+# Secretless Connector Plugins
+
+For general info on Secretless plugins, please refer to the [plugin README](../README.md).
+
+Secretless has built-in support for PostgreSQL, MySQL, APIs that use basic HTTP authentication, and many other target services.
+
+But what if you want to use Secretless with a target service it doesn't support out of the box?
+
+Service connector plugins let you extend Secretless to support any target service. If you know Go and understand
+your target service's authentication protocol, you can write a connector for it. It's easy and this guide walks you through it.
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/connector/README.md]: I agree. Also, `If you ... understand your target service's authentication protocol` is a BIG IF. So I hate to diminish the work it takes to create a new connector by saying `If <huge thing>, it's easy!`

File: pkg/secretless/plugin/connector/README.md [hunk 39]
```diff
@@ -0,0 +1,267 @@
...
+contributions as well. If you write a connector plugin you'd like to share with the community, please
+[let us know](https://secretless.io/community.html) or consider sending us a PR!
+
+## Technical Overview
+
+As described in the [plugin README](../README.md), Secretless uses [Go plugins](https://golang.org/pkg/plugin/)
+to produce shared object library files (with `.so` extensions) instead of a normal executable. The `.so` library files can
+be loaded at runtime to expand the number of target services your Secretless instance supports.
+
+Technically, a Secretless service connector plugin is just a Go shared library
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/connector/README.md]: I went with 
> Technically, a Secretless service connector plugin is a Go shared library file that implements two functions, either:

because it seemed clearer to me - please lmk if you object

File: pkg/secretless/plugin/connector/README.md [hunk 40]
```diff
@@ -0,0 +1,267 @@
...
+implementing.
+
+When Secretless runs, it will call `NewConnector` once, and then hold onto the
+returned `Connector`.  That `Connector` (remember: it's just a a single method)
+will then be called each time a new client connection requires authentication.
+
+Both `NewConnector` methods take only one argument -- `connector.Resources` --
+described below.
+
+The real work is done by the `Connector` functions they return...
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/connector/README.md]: it was suspense building! 😂 removing...

File: pkg/secretless/log/doc.go [hunk 41]
```diff
@@ -0,0 +1,35 @@
+/*
+Package log provides a type similar to the standard library's "log.Logger".  It allows you to
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/log/doc.go]: I was referring to the data `type` in go, but I think that's actually incorrect. The log package defines an _interface_.

File: pkg/secretless/log/doc.go [hunk 42]
```diff
@@ -0,0 +1,35 @@
...
+log events to stdout and stderr.  It respects Secretless's command line `debug` flag,
+so that calling Debugf or Infof will do nothing unless you started Secretless in
+debug mode.
+
+Your Secretless plugin code should never use the "fmt" or "log" packages or write
+directly to stdout or stderr.  Instead, call Logger() on your connector.Resources
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/log/doc.go]: Yeah, this was confusing. I took a stab at a rewrite.

File: pkg/secretless/log/doc.go [hunk 43]
```diff
@@ -0,0 +1,35 @@
...
+to get a secretless.Logger as defined below.
+
+The 3 Debug methods and the 3 Info methods do nothing unless you started
+Secretless with the Debug command line flag set to "true".  If you did start
+Secretless in debug mode, they write to stdout.
```
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/log/doc.go]: I looked over the file, and I _think_ my use is consistent. It's a little hard to tell, though.

godocs don't support code snippets inline with other text, so there is a lot of stuff here that would've been formatted as code that instead we're leveraging capitalization to indicate we're referring to method names.

File: pkg/secretless/plugin/connector/http/http.go (hunk 12 above)
Comment (izgeri) [reply pkg/secretless/plugin/connector/http/http.go]: it's referring to the specific bit of metadata in the `Authorization` header, so I think `Token` is correct - but I removed the backticks around Authorization to be consistent.

File: pkg/secretless/plugin/README.md (hunk 36 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/plugin/README.md]: Sounds good!

File: pkg/secretless/plugin/README.md (hunk 37 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/plugin/README.md]: That is a good way to approach it!

File: pkg/secretless/plugin/connector/README.md (hunk 38 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/plugin/connector/README.md]: Yes seems to be the biggest IF. Maybe separate knowing the authentication protocol from the same sentence as creating a new connector? Maybe list it a prerequisite note and then the remaining sentence can truthfully emphasize that the actual creation of the plugin is easy.

File: pkg/secretless/plugin/connector/README.md (hunk 40 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/plugin/connector/README.md]: Ah ok! Gotta have fun with it!

File: pkg/secretless/log/doc.go (hunk 41 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/log/doc.go]: Ok sounds good!

File: pkg/secretless/log/doc.go (hunk 42 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/log/doc.go]: Did you want me to review the rewrite or are you all set?

File: pkg/secretless/log/doc.go (hunk 43 above)
[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/log/doc.go]: I see what you mean - that's a perfectly good reason to use initial caps. I think it looked good otherwise and I'd agree with keeping the initial cap D here then.

File: pkg/secretless/plugin/connector/http/http.go (hunk 12 above)
Comment (suefran) [reply pkg/secretless/plugin/connector/http/http.go]: Sounds good!

**Comment by suefran** [comment]
@izgeri My edits to the PDFs are here. Any questions, let me know!

image

File: pkg/secretless/log/doc.go (hunk 42 above)
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/log/doc.go]: please do take a look, I'd like to double-check it makes more sense now

[the remark this answers is not in the document]
Comment (suefran) [reply pkg/secretless/log/doc.go]: Reads well now - thanks!

**Comment by izgeri** [comment]
@suefran I looked over the PDFs. I made some changes, but here are some notes:
- on the main page, `scheme` is correct. those docs are actually auto-generated.
- on the `http` page, I can't add another space after the `NewConnector` heading - it's treated differently because `NewConnector` is a method on the `ConnectorConstructor` type that's defined (so this is sort of like a subsection)

File: pkg/secretless/plugin/connector/README.md (hunk 38 above)
[the remark this answers is not in the document]
Comment (izgeri) [reply pkg/secretless/plugin/connector/README.md]: I think we'll need to add better notes on this, but I am ok with it not going in this PR. the team is thinking about this today, so once we get these changes in they can do some follow-on work to add more info on this piece. I will make sure you are notified about any of the follow-on changes.
